### ISSUE TITLE
Combine verification message and recovery message

### ISSUE THREAD

**Issue by kcole16** [near/near-wallet#622]
### Overview
We've had at least one instance a user triggering a resend of an email, which invalidated an earlier recovery email they'd successfully received (since we delete previous recovery keys during the resend), and then not receiving the new email. This can be catastrophic if it prevents the user from recovering their account.

One way to mitigate this is to combine the verification email (with a code) and the recovery email.

**Suggested Improvement**

After enabling either email or phone recovery, the user receives a single message with both a verification code and a recovery link. The user then enters the verification code into the UI (as is done now), and is instructed in both the message and UI to keep the message for future recovery.

Thoughts? @Patrick1904 @corwinharrell @mattlockyer

**Comment by mattlockyer** [comment]
If I understand, we're going to move the email of the recovery link into where we send the code, then verify the code in the UI and decide to either keep or reject this recovery key? Would there be a timeout on this, or is the wallet not created until I verify the code?

Repeated attempts would invalidate the existing key (that's presumably unverified) and resend code + recovery key.

The wallet shouldn't be created until I enter the correct code, because what if I've sent the code / recovery key to the wrong person (badly typed email). If the wallet is funded on creation, this opens up a window of attack.

**Comment by kcole16** [comment]
> The wallet shouldn't be created until I enter the correct code, because what if I've sent the code / recovery key to the wrong person (badly typed email).

Great point! Yes, we shouldn't add the key until after the correct code is entered.

Re: initial account creation, as part of #423, we should only create accounts once 2FA + recovery or Ledger initialization have happened. Which will mean post-email verification in cases where that is relevant.

**Comment by Patrick1904** [comment]
I'm not a big fan of even allowing simple re-send. Easy to just click the 're-send' button every time you want to log in on another device etc and ending up with tons of expired links. Happened to me. I think we should offer info behind a click (icon on profile) explaining how to find the recovery email if they're unable to locate it in their inbox. If the user is still unable to find the email, he/she can just disable it, and enable it again by going through the verification flow.

**Comment by corwinharrell** [comment]
@Patrick1904 if the user is already logged into their account, then that means they have already verified the email/phone they elected upon sign-up. What's the harm in allowing them to resend themselves the email if they have in fact lost/deleted their initial email?

Is your justification that it's almost easier to just hit "Resend email" than to go to the trouble of looking for the one you received when you initially created your account?

**Comment by Patrick1904** [comment]
Yes, security and perceived security is my main concern. It’s indeed too easy to click ‘resend’ instead of checking email. Lots of emails in the inbox leads to confusion (active/inactive links). Lots of emails makes it easier for potential attacker to identify sensitive data.

I’m assuming that most users that click ‘resend’ want to easily log in on another device. I think we should make adding another device super easy then by exposing temporary code/QR instead.
